pkg/apis/server.go

```diff
@@ -222,15 +235,39 @@ func serve(ctx context.Context, handler http.Handler, errorLog *stdlog.Logger, l
 	return nil
 }
 
-func newTcpListener(ctx context.Context, ip string, port int) (net.Listener, error) {
-	address := fmt.Sprintf("%s:%d", ip, port)
+func parseBindAddress(ip string, dual bool, port int) (network, address string, err error) {
```

Comment:
nit: The order `ip, port, dual` looks more comfortable to me.

Can we add some unit tests?

Reply:
updated